cloudmarker/events/rdbmsenforcetlsevent.py

```diff
@@ -0,0 +1,97 @@
+"""RDBMS Enforce TLS/SSL Event.
+
+This module defines the :class:`RDBMSEnforceTLSEvent` class that
+identifies RDBMS servers which have TLS/SSL connection enforcement
+disabled. This plugin works on the  properties found in the ``com``
+bucket of ``rdbms`` records.
+"""
+
+
+import logging
+
+from cloudmarker import util
+
+_log = logging.getLogger(__name__)
+
+
+class RDBMSEnforceTLSEvent:
+    """Az RDBMS TLS/SSL enforcement event plugin."""
+
+    def __init__(self):
+        """Create an instance of :class:`RDBMSEnforceTLSEvent`."""
+
+    def eval(self, record):
+        """Evaluate RDBMS servers for TLS connection enforcement.
+
+        Arguments:
+            record (dict): A RDBMS record.
+
+        Yields:
+            dict: An event record representing an RDBMS where TLS
+            connection enforcement is disabled
+
+        """
+        com = record.get('com', {})
+        if com is None:
+            return
+
+        if com.get('record_type') != 'rdbms':
+            return
+
+        ext = record.get('ext', {})
+        if ext is None:
+            return
+
+        if com.get('tls_enforced', True) is False:
```

Comment:
This is equivalent to the idiomatic form:

```
if not com.get('tls_enforced', True):
```

Reply:
I wrote a simple program and found out that what I am suggesting is indeed not equivalent to what you have written:

```python
# Solution 1.
com = {}
for flag_value in -1, True, None, False:
    com['tls_enforced'] = flag_value
    if flag_value == -1: # Missing key case.
        del com['tls_enforced']

    # Your correct solution.
    if com.get('tls_enforced', True) is False:
        print('bad')
    else:
        print('good')
print()


# Solution 2
com = {}
for flag_value in -1, True, None, False:
    com['tls_enforced'] = flag_value
    if flag_value == -1: # Missing key case.
        del com['tls_enforced']

    # My incorrect solution.
    if not com.get('tls_enforced', True):
        print('bad')
    else:
        print('good')
print()
```

Output:

```
good
good
good
bad

good
good
bad
bad
```

Having understood this, I still find specifying a default value of `True` in `com.get()` redundant. For example, see this code:

```python
# Solution 3
com = {}
for flag_value in -1, True, None, False:
    com['tls_enforced'] = flag_value
    if flag_value == -1: # Missing key case.
        del com['tls_enforced']

    # My suggestion: No default in com.get().
    if com.get('tls_enforced') is False:
        print('bad')
    else:
        print('good')
print()
```

Output:

```
good
good
good
bad
```

If the key is missing, `com.get()` would default to `None` anyway and you are already handing `None` as a good thing by the strict comparison with `is False`.

Since this part of the code took a while for me to understand, I think a comment would be very helpful for an occassionally dim-witted reader like me. 🙂 For example:

```python
    # True, None, or missing key is considered good. Only False is
    # considered a violation and leads to an event.
    if not com.get('tls_enforced') is False:
```

Reply:
Indeed, having a default value adds no value. I have changed the condition to: 
` if com.get('tls_enforced') is False:
`
and have added appropriate comments.